### PR TITLE
[APM-CI][Integration Tests] Fix Java integration tests

### PR DESCRIPTION
**Pull request by v1v**
This particular PR solves the integration test failures for the java agent.

## Highlights
- Move the docker `RUN` logic to the `maven-package.sh`
- Support release versions from the maven repository
- Transform `github;X.Y.Z` to `release;X.Y.Z`

### CODE REVIEW

File: docker/java/spring/maven-package.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  export JAVA_AGENT_BUILT_VERSION=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+fi
+
+cd /app
+mvn -q --batch-mode -DAGENT_API_VERSION=${JAVA_AGENT_BUILT_VERSION} \
+  -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+  package
+
+cp /root/.m2/repository/co/elastic/apm/apm-agent-api/${JAVA_AGENT_BUILT_VERSION}/apm-agent-api-${JAVA_AGENT_BUILT_VERSION}.jar /agent/apm-agent.jar
```
Comment (felixbarny) [inline docker/java/spring/maven-package.sh]: This is the jar for the public api, not the actual agent. the agent is in `elastic-apm-agent/${JAVA_AGENT_BUILT_VERSION}/elastic-apm-agent-${JAVA_AGENT_BUILT_VERSION}.jar`

Comment (kuisathaverat) [reply docker/java/spring/maven-package.sh]: 🤦‍♂️

**Comment by felixbarny** [comment]
is that also a fix for https://github.com/elastic/apm-integration-testing/issues/218?

File: docker/java/spring/maven-package.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+JAVA_AGENT_BUILT_VERSION=${1}
+
+ARTIFACT_ID=elastic-apm-agent
+
+if [ -z "${JAVA_AGENT_BUILT_VERSION}" ] ; then
+  cd /agent/apm-agent-java
+  mvn -q --batch-mode install -DskipTests \
+      -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
+
+  export JAVA_AGENT_BUILT_VERSION=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
```
Comment (felixbarny) [inline docker/java/spring/maven-package.sh]: Is there a way to not specify the version of the plugin? It just seems a bit brittle.

Comment (v1v) [reply docker/java/spring/maven-package.sh]: Do you mean `org.codehaus.mojo:exec-maven-plugin:1.3.1:exec`? If so, it's safety using the specific version to ensure there are no regressions in the plugin itself with new versions.

Comment (felixbarny) [reply docker/java/spring/maven-package.sh]: Ah, gotcha

**Comment by v1v** [comment]
jenkins run the tests

**Comment by kuisathaverat** [comment]
> is that also a fix for #218?

probably it is resolved, both ARGs are always used before this change

**Pull request merged**